### PR TITLE
refactor(rn): clean things up so they're more digestable

### PR DESCRIPTION
**Pull request by moldy530**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR focuses on refactoring the `account-kit/rn-signer` codebase, enhancing the `RNSignerClient` functionality, and updating the Android integration for better key management and security protocols. It also modifies example usage and improves configuration files.

### Detailed summary
- Removed `NativeTEKStamper` and replaced with `RNSignerClient`.
- Updated `package.json` dependencies.
- Added new files for HPKE key management in Android.
- Enhanced `.env` setup instructions in `README.md`.
- Modified Metro configuration to include additional watch folders.
- Updated `yarn.lock` with new package versions.
- Refactored methods in `NativeTEKStamperModule` for better clarity and security.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by moldy530** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1134?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1134** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1134?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1132** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1132?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1131** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1131?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1130** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1130?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1129** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1129?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1128** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1128?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @moldy530 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by dphilipson — APPROVED** [review]
Looks great! Some Kotlin nitpicks, but should be good to go.

File: account-kit/rn-signer/android/src/main/java/com/accountkit/reactnativesigner/TEKManager.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        cipherText: ByteArray,
+        info: ByteArray,
+        aad: ByteArray
+    ): ByteArray {
+        // Why do we hve to do all this rather than doing:
+        // val hybridDecrypt = tekHandle.getPrimitive(HybridDecrypt::class.java)
+        // val decryptedKey = hybridDecrypt.decrypt(ciphertext, "turnkey_hpke".toByteArray())
+        // the hybridDecrypt.decrypt that google exposes doesn't allow us to pass in
+        // the aad that's needed to complete decryption
+        val keyHandle = getKeysetHandle() ?: throw Exception("No TEK found!")
```
[the remark this answers is not in the document]
Comment (moldy530) [reply account-kit/rn-signer/android/src/main/java/com/accountkit/reactnativesigner/TEKManager.kt:34]: oh didn't know about that syntax will give it a try. This syntax is what android studio recommended when I had it in an `if` block

[the remark this answers is not in the document]
Comment (dphilipson) [reply account-kit/rn-signer/android/src/main/java/com/accountkit/reactnativesigner/TEKManager.kt:34]: yeah, that syntax looks good too, no need to change it.

**Pull request merged**